Dockerfile

```diff
@@ -60,19 +68,22 @@ RUN apt-add-repository ppa:brightbox/ruby-ng \
   && apt-get install -y ruby2.6 ruby2.6-dev \
   && gem update --system 3.2.14 \
   && gem install bundler -v 1.17.3 --no-document \
-  && gem install bundler -v 2.2.15 --no-document
-
+  && gem install bundler -v 2.2.15 --no-document \
+  && rm -Rf /var/lib/gems/2.6.0/cache/*
 
 ### PYTHON
 
 # Install Python 2.7 and 3.9 with pyenv. Using pyenv lets us support multiple Pythons
 ENV PYENV_ROOT=/usr/local/.pyenv \
   PATH="/usr/local/.pyenv/bin:$PATH"
+RUN mkdir -p "$PYENV_ROOT" && chown dependabot:dependabot "$PYENV_ROOT"
+USER dependabot
 RUN git clone https://github.com/pyenv/pyenv.git /usr/local/.pyenv \
   && cd /usr/local/.pyenv && git checkout 1.2.26 && cd - \
   && pyenv install 3.9.4 \
   && pyenv install 2.7.18 \
   && pyenv global 3.9.4
```

Comment:
I have come to greatly appreciate [asdf](https://asdf-vm.com/#/).

If we create a `.tool-versions` file with the following:

```plaintext
python 3.9.4 2.7.18
ruby 2.7.2
golang 1.16.2
rust 1.50.0
```

Then we can install multiple versions of tools via `asdf install` assuming we've installed the appropriate plugins. Writing plugins is very easy as well so we could swap out the default plugins for our own custom versions that are optimized for this runtime environment.

Some of the plugins also support `.default-<packages` configs to install packages for each version of that tool.

E.g.

```bash
モ cat ~/.default-gems
bundler
interactive_editor
jive
neovim
solargraph
spandx

モ cat ~/.default-python-packages
awscli
docker-compose
lookatme
pass-git-helper
pip
pipenv
turses
twine
wheel
```

Reply:
Nice! I haven't had a chance to play with that yet but looks promising! Seems like a good follow up to investigate 🙌  do you know if it can install things from binaries or builds from source?